ciphers/porta_cipher.py

```diff
@@ -0,0 +1,113 @@
+    "Y": [["A", "B", "C", "D", "E", "F", "G", "H", "I", "J", "K", "L", "M"],
+          ["O", "P", "Q", "R", "S", "T", "U", "V", "W", "X", "Y", "Z", "N"]], "Z": [["A", "B", "C", "D", "E", "F", "G", "H", "I", "J", "K", "L", "M"], ["O", "P", "Q", "R", "S", "T", "U", "V", "W", "X", "Y", "Z", "N"]]}
+
+
+def generateTable(key):
+    """
+    >>> generateTable('marvin')
+    [[['A', 'B', 'C', 'D', 'E', 'F', 'G', 'H', 'I', 'J', 'K', 'L', 'M'], ['U', 'V', 'W', 'X', 'Y', 'Z', 'N', 'O', 'P', 'Q', 'R', 'S', 'T']], [['A', 'B', 'C', 'D', 'E', 'F', 'G', 'H', 'I', 'J', 'K', 'L', 'M'], ['N', 'O', 'P', 'Q', 'R', 'S', 'T', 'U', 'V', 'W', 'X', 'Y', 'Z']], [['A', 'B', 'C', 'D', 'E', 'F', 'G', 'H', 'I', 'J', 'K', 'L', 'M'], ['S', 'T', 'U', 'V', 'W', 'X', 'Y', 'Z', 'N', 'O', 'P', 'Q', 'R']], [['A', 'B', 'C', 'D', 'E', 'F', 'G', 'H', 'I', 'J', 'K', 'L', 'M'], ['Q', 'R', 'S', 'T', 'U', 'V', 'W', 'X', 'Y', 'Z', 'N', 'O', 'P']], [['A', 'B', 'C', 'D', 'E', 'F', 'G', 'H', 'I', 'J', 'K', 'L', 'M'], ['W', 'X', 'Y', 'Z', 'N', 'O', 'P', 'Q', 'R', 'S', 'T', 'U', 'V']], [['A', 'B', 'C', 'D', 'E', 'F', 'G', 'H', 'I', 'J', 'K', 'L', 'M'], ['U', 'V', 'W', 'X', 'Y', 'Z', 'N', 'O', 'P', 'Q', 'R', 'S', 'T']]]
+    """
+    table = []
+    for char in key.upper():
+        table.append(alphabet[char])
+    return table
+
+
+def encrypt(key, words):
+    """
+    >>> encrypt('marvin', 'jessica')
+    'QRACRWU'
+    """
+    cipher = ""
+    count = 0
+    table = generateTable(key)
+    for char in words.upper():
+        cipher += getOpponent(table[count], char)
+        count = (count + 1) % len(table)
+    return cipher
+
+
+def decrypt(key, words):
+    """
+    >>> decrypt('marvin', 'QRACRWU')
+    'JESSICA'
+    """
+    return encrypt(key, words)
+
+
+def getPosition(table, char):
+    """
+    >>> table = [[['A', 'B', 'C', 'D', 'E', 'F', 'G', 'H', 'I', 'J', 'K', 'L', 'M'], ['U', 'V', 'W', 'X', 'Y', 'Z', 'N', 'O', 'P', 'Q', 'R', 'S', 'T']], [['A', 'B', 'C', 'D', 'E', 'F', 'G', 'H', 'I', 'J', 'K', 'L', 'M'], ['N', 'O', 'P', 'Q', 'R', 'S', 'T', 'U', 'V', 'W', 'X', 'Y', 'Z']], [['A', 'B', 'C', 'D', 'E', 'F', 'G', 'H', 'I', 'J', 'K', 'L', 'M'], ['S', 'T', 'U', 'V', 'W', 'X', 'Y', 'Z', 'N', 'O', 'P', 'Q', 'R']], [['A', 'B', 'C', 'D', 'E', 'F', 'G', 'H', 'I', 'J', 'K', 'L', 'M'], ['Q', 'R', 'S', 'T', 'U', 'V', 'W', 'X', 'Y', 'Z', 'N', 'O', 'P']], [['A', 'B', 'C', 'D', 'E', 'F', 'G', 'H', 'I', 'J', 'K', 'L', 'M'], ['W', 'X', 'Y', 'Z', 'N', 'O', 'P', 'Q', 'R', 'S', 'T', 'U', 'V']], [['A', 'B', 'C', 'D', 'E', 'F', 'G', 'H', 'I', 'J', 'K', 'L', 'M'], ['U', 'V', 'W', 'X', 'Y', 'Z', 'N', 'O', 'P', 'Q', 'R', 'S', 'T']]]
+    >>> getPosition(table, 'A')
+    (None, None)
+    """
+    row = -1
+
+    if char in table[0]:
+        row = 0
+    elif char in table[1]:
+        row = 1
+
+    if row != -1:
+        return (row, table[row].index(char))
+    else:
+        return (None, None)
```

Comment:
Please convert these four lines into a one liner using a ternary if.
`return (None, None) if row == -1 else (row, table[row].index(char))`

Reply:
updated in latest